src/specialized.jl

```diff
@@ -58,7 +58,7 @@ function predict_string(ps::MemoryParserState)
             if ps.utf8data[s] == LATIN_U  # Unicode escape
                 t = ps.s
                 ps.s = s + 1
-                len += length(string(read_unicode_escape!(ps)).data)
+                len += sizeof(string(read_unicode_escape!(ps)))
```

Comment:
The performance of Unicode escapes is a bit of a marginal use case, but you're right. It would be nice to have a package with functions like these for working with UTF8. (this also applies to the Char-encoding logic)